src/transformers/pipelines/__init__.py

```diff
@@ -425,10 +435,24 @@ def pipeline(
                 "Trying to load the model with Tensorflow."
             )
 
+        if task == "automatic-speech-recognition" and "feature_extractor" not in kwargs:
+            feature_extractor = AutoFeatureExtractor.from_pretrained(model)
+            kwargs["feature_extractor"] = feature_extractor
+
+        # Make sure the model class becomes iterable
         if model_class is None:
             raise ValueError(
                 f"Pipeline using {framework} framework, but this framework is not supported by this pipeline."
             )
+        elif model_class == "config":
+            if config is None:
+                config = AutoConfig.from_pretrained(model, revision=revision, _from_pipeline=task, **model_kwargs)
+            try:
+                import transformers
+
+                model_class = getattr(transformers, config.architectures[0])
+            except Exception:
+                raise Exception(f"Could not load default model class for {model}")
```

Comment:
I'm not super fan of this. I think this is a workaround the fact there is not proper `AutoModel` class for the `"automatic-speech-recognition"` which should be the case (as you pointed out in the conversation).

Reply:
Completely agree with you here ! I am not a fan either, but it seemed like the simplest possible change currently.

However the 2 classes *are* quite different:
- Wav2vec2 does ASR in a single pass,
- S2T needs to run generate.
Hence this: https://github.com/huggingface/transformers/blob/master/src/transformers/pipelines/automatic_speech_recognition.py#L141

I really think there is no way to avoid it thinking about what `ForConditionalGeneration` and `ForCTC`  to fuse them in some sort of common `For` class. The heads are very different and should be used differently.
There are also differences in the tokenizers (in CTC, `pad` is used to split multiple letters, whereas you want to remove `pad`  in S2T).

We could also imagine having 2 different pipelines, but it feels clunky too because the 2 pipelines API are exactly the same.

Reply:
Pinging @patrickvonplaten

Reply:
I didn't make an AutoClass yet since it's so different. IMO, we should add `S2T` to `AutoModelForConditionalGeneration` since that's exactly what it does and for Wav2Vec2 we should create a new AutoClass like, `AutoModelForCTC`